pwa/src/components/applications/applicationForm.tsx

```diff
@@ -149,6 +151,7 @@ export const ApplicationForm: React.FC<ApplicationFormProps> = ({id}) => {
                     <Spinner/>
                   ) : (
                     <div>
+                          {overlaySpinner && <LoadingOverlay /> }
```

Comment:
This is the easiest spot to point this out: it might be best to choose a consistent name for both the `state` and the `component`. You're now using `overlaySpinner` as `state` and `LoadingOverlay` as `component` name.

This would be much more clear when we do one of the following:

1. Change the `state` variable in all files to `loadingOverlay` (and its setter to `setLoadingOverlay`) or
2. Change the `component` name to `OverlaySpinner`.

That way, you get a much more clearer line here (and in the other files, like e.g.):

`{ loadingOverlay && setLoadingOverlay }`

Another `state` variable that would make sense would just be `loading` and `setLoading` so your syntax could become `loading && <LoadingOverlay />`.

Doesn't really matter which one you choose, but I think implementing one of these suggestions would be a good addition 👍🏼